lib/plugins/entry-files-manifest.js

```diff
@@ -13,32 +13,66 @@ const PluginPriorities = require('./plugin-priorities');
 const sharedEntryTmpName = require('../utils/sharedEntryTmpName');
 const copyEntryTmpName = require('../utils/copyEntryTmpName');
 const AssetsPlugin = require('assets-webpack-plugin');
+const fs = require('fs');
+const path = require('path');
+const crypto = require('crypto');
 
-function processOutput(assets) {
-    for (const entry of [copyEntryTmpName, sharedEntryTmpName]) {
-        delete assets[entry];
-    }
-
-    // with --watch or dev-server, subsequent calls will include
-    // the original assets (so, assets.entrypoints) + the new
-    // assets (which will have their original structure). We
-    // delete the entrypoints key, and then process the new assets
-    // like normal below
-    delete assets.entrypoints;
-
-    // This will iterate over all the entry points and convert the
-    // one file entries into an array of one entry since that was how the entry point file was before this change.
-    for (const asset in assets) {
-        for (const fileType in assets[asset]) {
-            if (!Array.isArray(assets[asset][fileType])) {
-                assets[asset][fileType] = [assets[asset][fileType]];
+function processOutput(webpackConfig) {
+    return (assets) => {
+        for (const entry of [copyEntryTmpName, sharedEntryTmpName]) {
+            delete assets[entry];
+        }
+
+        // with --watch or dev-server, subsequent calls will include
+        // the original assets (so, assets.entrypoints) + the new
+        // assets (which will have their original structure). We
+        // delete the entrypoints key, and then process the new assets
+        // like normal below
+        delete assets.entrypoints;
+
+        // This will iterate over all the entry points and convert the
+        // one file entries into an array of one entry since that was how the entry point file was before this change.
+        const integrity = {};
+        const integrityAlgorithm = webpackConfig.integrityAlgorithm;
+        const publicPath = webpackConfig.getRealPublicPath();
+
+        if (integrityAlgorithm) {
+            integrity.algorithm = integrityAlgorithm;
+            integrity.hashes = {};
+        }
+
+        for (const asset in assets) {
+            for (const fileType in assets[asset]) {
+                if (!Array.isArray(assets[asset][fileType])) {
+                    assets[asset][fileType] = [assets[asset][fileType]];
+                }
+
+                if (webpackConfig.integrityAlgorithm) {
+                    for (const file of assets[asset][fileType]) {
+                        if (!(file in integrity.hashes)) {
```

Comment:
Is file sometimes already in `integrity.hashes`? Like, maybe a shared file that's used by multiple entries?

Also, we can flip this to `if (file in integrity.hashes) { continue; }

Reply:
> Is file sometimes already in `integrity.hashes`? Like, maybe a shared file that's used by multiple entries?

Yep, the `runtime.js` file for instance :)

> Also, we can flip this to `if (file in integrity.hashes) { continue; }

Will be part of the next changeset!